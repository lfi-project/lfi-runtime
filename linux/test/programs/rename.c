#include <stdio.h>
#include <errno.h>

int main(int argc, char **argv) {
    (void) argc;
    const char *original_name = argv[1];
    const char *temp_name = "temp_file.txt";

    // Rename original_file.txt to temp_file.txt
    if (rename(original_name, temp_name) == 0) {
        printf("Renamed '%s' to '%s'.\n", original_name, temp_name);
    } else {
        perror("rename to temp_name");
        return 1;
    }

    // Rename temp_file.txt back to original_file.txt
    if (rename(temp_name, original_name) == 0) {
        printf("Renamed '%s' back to '%s'.\n", temp_name, original_name);
    } else {
        perror("rename back to original_name");
        return 1;
    }

    return 0;
}
