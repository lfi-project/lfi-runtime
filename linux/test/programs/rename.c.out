Renamed 'programs/inputs/rename.txt' to 'temp_file.txt'.
Renamed 'temp_file.txt' back to 'programs/inputs/rename.txt'.
